packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1236,7 +1245,15 @@ function createChildReconciler(
     if (oldFiber === null) {
       // If we don't have any more existing children we can choose a fast path
       // since the rest will all be insertions.
-      for (; !step.done; newIdx++, step = newChildren.next()) {
+      for (
+        ;
+        !step.done;
+        newIdx++,
+          step = newChildren.next(),
+          knownKeys = __DEV__
+            ? warnOnInvalidKey(step.value, knownKeys, returnFiber)
+            : null
```

Comment:
Might DCE better if you move this to the first line of the for block. It's not big but it leaves behind a null in the prod builds for each instance

Reply:
The way this is structured, that doesn't mean the same thing because this should only execute at the beginning of the next iteration loop - not the first one. A proper structure would need to change the way the `continue` calls are placed.